Definition/Type/CustomScalarType.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Overblog\GraphQLBundle\Definition\Type;
+
+use GraphQL\Type\Definition\CustomScalarType as BaseCustomScalarType;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Utils\Utils;
+
+class CustomScalarType extends BaseCustomScalarType
+{
+    public function __construct(array $config = [])
+    {
+        $config['name'] = isset($config['name']) ? $config['name'] : uniqid('CustomScalar');
+        parent::__construct($config);
+
+        $this->config['scalarType'] = isset($this->config['scalarType']) ? $this->config['scalarType'] : null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function serialize($value)
+    {
+        return $this->call('serialize', $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parseValue($value)
+    {
+        return $this->call('parseValue', $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parseLiteral(/* GraphQL\Language\AST\ValueNode */ $valueNode)
+    {
+        return $this->call('parseLiteral', $valueNode);
+    }
+
+    private function call($type, $value)
+    {
+        if (isset($this->config['scalarType'])) {
+            $scalarType = $this->config['scalarType'];
+            $scalarType = is_callable($scalarType) ? $scalarType() : $scalarType;
```

Comment:
Why do you reassign `$scalarType`?

Reply:
Because we also allow lazy loading of the `scalarType` config entry.